ContentModification/EmbedFonts/EmbedFonts.cpp

```diff
@@ -1,194 +1,217 @@
             }
+
+            PDEFontSubsetNow(pdeFontUnembedded, PDDocGetCosDoc(pdDoc));
+
+            PDERelease((PDEObject)pdeFont);
         }
+        else {
+            // Fully embed font
+            PDEFont pdeFont = PDEFontCreateFromSysFont(sysFont, kPDEFontCreateEmbedded);
+            PDEFontEmbedNow(pdeFontUnembedded, PDDocGetCosDoc(pdDoc));
 
-        if (sysEnc)
-            PDERelease((PDEObject)sysEnc);
+            PDERelease((PDEObject)pdeFont);
+        }
+    }
 
+    if (sysEnc)
+    {
+        PDERelease((PDEObject)sysEnc);
+    }
     HANDLER
         APDFLib::displayError(ERRORCODE);
     END_HANDLER
 }
 
-ACCB1 ASBool ACCB2 GetFontInfoProc(PDFont pdFont, PDFontFlags *pdFontFlagsPtr, void *clientData) {
+ACCB1 ASBool ACCB2 GetFontInfoProc(PDFont pdFont, PDFontFlags* fontFlags, std::vector<UnembeddedFont>& clientData) {
     CosObj cosFont;
     PDEFontAttrs attrs;
     PDSysFont sysFont;
     char fontNameBuf[PSNAMESIZE];
-    const char *fontSubtypeP;
-    char *fontNameStart = 0;
+    const char* fontSubtypeP;
+    char* fontNameStart = 0;
     ASBool fontEmbedded = false, fontSubset = false, fontIsSysFont = false;
     DURING
-        memset(&attrs, 0, sizeof(attrs));
-
-        PDFontGetName(pdFont, fontNameBuf, PSNAMESIZE);
-        attrs.name = ASAtomFromString(fontNameBuf);
-        attrs.type = PDFontGetSubtype(pdFont);
-        fontSubtypeP = ASAtomGetString(attrs.type);
-
-        fontEmbedded = PDFontIsEmbedded(pdFont);
-        // Subset test: a font was subset if the 7th character is '+' (a plus-sign).
-        if (fontEmbedded) {
-            if ((strlen(fontNameBuf)) > 7 && (fontNameBuf[6] == '+'))
-                fontSubset = true;
-        }
-        if (fontSubset)
-            fontNameStart = fontNameBuf + 7; // skip the "ABCDEF+"
-        else
-            fontNameStart = fontNameBuf;
-
-        // If the font is not found on the system, sysFont will be 0.
-        sysFont = PDFindSysFont(&attrs, sizeof(PDEFontAttrs), 0);
-        if (sysFont)
-            fontIsSysFont = true;
-
-        // Print font information
-        std::cout << "Font " << fontNameStart << ", Subtype " << fontSubtypeP << " ("
-                  << (fontIsSysFont ? "" : "Not a ") << "System Font, ";
-        if (fontEmbedded) {
-            std::cout << "embedded" << (fontSubset ? " subset" : "");
-        } else {
-            std::cout << "unembedded";
-        }
-        std::cout << ")" << std::endl;
-
-        // Add font to the list of fonts to be subset. This example only subsets System
-        // fonts that are not embedded in this PDF document. The sample will not subset fonts
-        // that are fully embedded in the PDF file.
-        if (fontIsSysFont && !fontEmbedded) {
-            pContainerOfFonts pcf = (pContainerOfFonts)clientData;
-            cosFont = PDFontGetCosObj(pdFont);
-            pcf->push_back(new _t_pdfUsedFont(cosFont));
-        }
+        memset(&attrs, 0, sizeof(PDEFontAttrs));
```

Comment:
The indentation is different from the lines below it.

Reply:
fixed